aws-rds-dbinstance/src/main/java/software/amazon/rds/dbinstance/Translator.java

```diff
@@ -303,15 +303,19 @@ public static ModifyDbInstanceRequest modifyDbInstanceRequest(
                 .promotionTier(desiredModel.getPromotionTier())
                 .storageType(desiredModel.getStorageType())
                 .tdeCredentialArn(desiredModel.getTdeCredentialArn())
-                .tdeCredentialPassword(desiredModel.getTdeCredentialPassword())
-                .vpcSecurityGroupIds(desiredModel.getVPCSecurityGroups());
+                .tdeCredentialPassword(desiredModel.getTdeCredentialPassword());
 
         final CloudwatchLogsExportConfiguration cloudwatchLogsExportConfiguration = buildTranslateCloudwatchLogsExportConfiguration(
                 Optional.ofNullable(previousModel).map(ResourceModel::getEnableCloudwatchLogsExports).orElse(Collections.emptyList()),
                 desiredModel.getEnableCloudwatchLogsExports()
         );
         builder.cloudwatchLogsExportConfiguration(cloudwatchLogsExportConfiguration);
 
+        // VPC security groups can not be assigned directly on an Aurora instance.
+        if (StringUtils.isBlank(desiredModel.getDBClusterIdentifier())) {
+            builder.vpcSecurityGroupIds(desiredModel.getVPCSecurityGroups());
+        }
+
```

Comment:
I am afraid this can be misleading in case of customer specified `DBClusterIdentifier` and `vpcSecurityGroupIds` in update. `vpcSecurityGroupIds` will be ignored and update succeed instead of failure as customer can not update `vpcSecurityGroupIds` if db instance is part of cluster.